src/queue.c

```diff
@@ -3945,10 +3958,10 @@ _dispatch_runloop_queue_poke(dispatch_queue_t dq, dispatch_qos_t qos,
 
 DISPATCH_NOINLINE
 static void
-_dispatch_global_queue_poke_slow(dispatch_queue_t dq, unsigned int n)
+_dispatch_global_queue_poke_slow(dispatch_queue_t dq, int n, int floor)
```

Comment:
@das this is where the part I'd like you to have a look at starts, to the end of the file. For Darwin `floor` will always be 0. For Linux it can be a negative value that allows to oversubscribe the pool further.